homeassistant/components/switch/rpi_rf.py

```diff
@@ -109,9 +113,10 @@ def is_on(self):
 
     def _send_code(self, code_list, protocol, pulselength):
         """Send the code(s) with a specified pulselength."""
-        _LOGGER.info("Sending code(s): %s", code_list)
-        for code in code_list:
-            self._rfdevice.tx_code(code, protocol, pulselength)
+        with self._lock:
+            _LOGGER.info("Sending code(s): %s", code_list)
+            for code in code_list:
+                self._rfdevice.tx_code(code, protocol, pulselength)
         return True
```

Comment:
Also old code, but this doesn't seem to change anything, since it always returns `True`.